### PR TITLE
[Filesystem] keep perms when copying

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #13222 |
| License | MIT |
| Doc PR | - |

Replaces PR #13222

### CODE REVIEW

**Comment by fabpot** [comment]
IIUC, the goal of the PR that was merged was to mimic the behavior of `cp` where only the executable bit is kept when copying.

**Comment by nicolas-grekas** [comment]
updated

File: src/Symfony/Component/Filesystem/Tests/FilesystemTest.php [hunk 1]
```diff
@@ -996,29 +996,16 @@ public function testDumpFileOverwritesAnExistingFile()
         $this->assertSame('bar', file_get_contents($filename));
     }
 
-    public function testCopyShouldKeepExecutionPermission()
+    public function testCopyShouldKeepPermissions()
```
Comment (fabpot) [inline src/Symfony/Component/Filesystem/Tests/FilesystemTest.php]: So, the previous name was more accurate, right?

Comment (nicolas-grekas) [inline src/Symfony/Component/Filesystem/Tests/FilesystemTest.php]: Yep, fixed

File: src/Symfony/Component/Filesystem/Filesystem.php [hunk 2]
```diff
@@ -69,10 +69,7 @@ public function copy($originFile, $targetFile, $override = false)
                 throw new IOException(sprintf('Failed to copy "%s" to "%s".', $originFile, $targetFile), 0, null, $originFile);
             }
 
-            if (is_executable($originFile)) {
-                // User Executable | Group Executable | Other Executable
-                chmod($targetFile, fileperms($targetFile) | 0x0040 | 0x0008 | 0x0001);
-            }
+            @chmod($targetFile, fileperms($targetFile) | (fileperms($originFile) & 0111));
```
Comment (fabpot) [inline src/Symfony/Component/Filesystem/Filesystem.php]: What about adding a comment above explaining what it does and why we only keep the executable bit?

Comment (nicolas-grekas) [inline src/Symfony/Component/Filesystem/Filesystem.php]: comment added

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**